src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFeatures.java

```diff
@@ -578,12 +578,12 @@ public void hitAnchorTp(PlayerInteractEvent event) {
 			return;
 		}
 		for (int y = block.getY() + 1; y <= 255; y++) {
-			if (doTeleport(block, event.getPlayer(), y)) {
+			if (event.getAction() == Action.RIGHT_CLICK_BLOCK && doTeleport(block, event.getPlayer(), y)) {
```

Comment:
You should decide based on the type of click which loop you use, not do both loops and check inside of them every time. Right now you're doing the event.getAction() comparison 255 times worst case when you should only need to do it once